### DISCUSSION TITLE
why shouldnt OTD run as admin?

### DISCUSSION THREAD

**Discussion by aplotor** [OpenTabletDriver/OpenTabletDriver#2064] [Q&A] [answered]
well first of all im not using the built in admin account, and i didnt set OTD to run as admin. when i launch OTD it says OTD should not be run with admin privileges etc. genuine question, why not? and why would some features not work as intended if it's run as admin?

also, since im not using the built in admin account and i didnt set OTD to run as admin, i guess the only thing left to "fix" this is to turn off UAC, but i like having UAC off and dont want to turn it on just for OTD.

**Comment by InfinityGhost** [comment, marked as answer]
> when i launch OTD it says OTD should not be run with admin privileges etc. genuine question, why not? and why would some features not work as intended if it's run as admin?

Short answer: Running as admin completely breaks WPF dialogs because of an upstream dependency issue (afaik)

Long answer: Running anything as admin is an inherent security risk and being that OpenTabletDriver has network access via the plugin manager, the updater, etc. its best to just not support it as admin in the first place.

> also, since im not using the built in admin account and i didnt set OTD to run as admin, i guess the only thing left to "fix" this is to turn off UAC, but i like having UAC off and dont want to turn it on just for OTD.

The only fix is to use UAC. There is a reason UAC exists, as its the only thing stopping malware from having full access to everything on your PC. Running with UAC disabled is the *nix equivalent of running as the root account at all times.

UAC is certainly not perfect, but its certainly more secure than running without it.
